cmd/kube-scheduler/app/testing/testserver.go

```diff
@@ -111,14 +113,16 @@ func StartTestServer(t Logger, customFlags []string) (result TestServer, err err
 		s.CombinedInsecureServing.Metrics.Listener = listener
 		t.Logf("kube-scheduler will listen insecurely on port %d...", s.CombinedInsecureServing.BindPort)
 	}
-	config, err := s.Config()
+
+	cc, sched, err := app.SetupCommand(ctx, []string{}, s)
 	if err != nil {
 		return result, fmt.Errorf("failed to create config from options: %v", err)
 	}
+	config := cc.Config
```

Comment:
having both `cc` and `config` is confusing. Prefer to use cc in the remaining lines.
The only exception is probably `result.Config = cc.Config`